katcp/server.py

```diff
@@ -2325,7 +2326,7 @@ def set_log_level(self, level):
         """
         self._log_level = level
         if self._python_logger:
-            level = PYTHON_LEVEL.get(level)
+            level = self.PYTHON_LEVEL.get(level)
```

Comment:
On second thought ... you could just index using [ ] and then catch the ValueError and report a more proper error message so that you know why KATCP blows up. That is how `def set_log_level_by_name(self, level_name)` does it.